source/common/http/http2/codec_impl.cc

```diff
@@ -487,6 +487,10 @@ void ConnectionImpl::StreamImpl::encodeDataHelper(Buffer::Instance& data, bool e
   auto status = parent_.sendPendingFrames();
   // See comment in the `encodeHeadersBase()` method about this RELEASE_ASSERT.
   RELEASE_ASSERT(status.ok(), "sendPendingFrames() failure in non dispatching context");
+  if (!parent_.protocol_constraints_.checkOutboundFrameLimits().ok()) {
```

Comment:
Yes, it is possible. Once case I can think of is receiving multiple DATA frames from the upstream server. Since we do not propagate errors between codecs, the upstream codec will continue to proxy DATA frames even though downstream frame queue is flooded. This is existing behavior. We could add error propagation between codecs later on to optimize this.

Reply:
Should we avoid the call to sendPendingFrames if an error was already set prior to getting there?

I'm also confused about sendPendingFrames returning a non-error status while protocol_constraints_.checkOutboundFrameLimits contains an error.

Reply:
I think such change would be too risky. `sendPendingFrames()` may cause nghttp2 callbacks to be called and if we skip it some internal state may not be updated. The idea for handling errors in the non dispatching context is to continue as normal and let I/O event complete and then disconnect in a separate event.